### PR TITLE
Better status action

### PR DESCRIPTION
**Pull request by evantahler**
Add more data

- include event loop delay
- include memory consumption (HEAP)
- total resque queue length

Return a "healthy" or "unhealthy"  response based on this new data (and configurable via ENV)

### CODE REVIEW

File: actions/status.js [hunk 1]
```diff
@@ -9,10 +17,69 @@ exports.status = {
   },
 
   run: function(api, data, next){
+
+    /* --- Define Helpers --- */
+
+    var checkRam = function(callback){
```
Comment (gcoonrod) [inline actions/status.js]: Why not move these helper functions into `api.status` initializer? Provide a couple OOTB status apis.

File: actions/status.js [hunk 2]
```diff
@@ -9,10 +17,69 @@ exports.status = {
...
+
     data.response.id                = api.id;
     data.response.actionheroVersion = api.actionheroVersion;
     data.response.uptime            = new Date().getTime() - api.bootTime;
+    data.response.name              = packageJSON.name;
+    data.response.description       = packageJSON.description;
+    data.response.version           = packageJSON.version;
 
-    next();
+    checkRam(function(error){
```
Comment (gcoonrod) [inline actions/status.js]: No async series?

File: initializers/utils.js [hunk 3]
```diff
@@ -244,6 +245,32 @@ module.exports = {
       return {host: host, port: parseInt(port, 10)};
     };
 
+    api.utils.eventLoopDelay = function(itterations, callback){
+      var intervalJobs = [];
+      var intervalTimes = [];
+
+      var i = 0;
+      while(i < itterations){
```
Comment (gcoonrod) [inline initializers/utils.js]: Should we do a check on `itterations` to avoid any endless while loops?

File: actions/status.js (hunk 1 above)
Comment (evantahler) [inline actions/status.js]: I did that at first (and started storing samples), but the more I thought about it, the more I realized that what you consider "important" for health varies drastically between applications.  To that end, I didn't want to add *any* initializers for this, but to showcase *some* of the things you might check, and how to do it.

That'' also why I'm using ENV here and not configuration varialbes.  I expect this file to be drastically changed in a production project.

File: actions/status.js (hunk 2 above)
Comment (evantahler) [inline actions/status.js]: this needs to work in new projects which won't have async installed by default.

File: actions/status.js (hunk 1 above)
Comment (gcoonrod) [inline actions/status.js]: True, but I think that AH should have a minimum opinion about whether or not an individual node is healthy. Now, you could achieve that with documentation. But I personally think that adding a few initializers is valuable.

Comment (evantahler) [inline actions/status.js]: I disagree.  For example, the event-loop check seems like a sefe check for a node that is only serving HTTP traffic.  However, if you have a node which is only doing tasks, it's totally fine to block the event loop... and if you are doing something like image processing, that's expected behavior.

File: initializers/utils.js (hunk 3 above)
Comment (evantahler) [inline initializers/utils.js]: yep! https://github.com/evantahler/actionhero/pull/895/commits/def9c9d97f1a48fd9a8e2b1decfd9ac06a64e3c9

File: initializers/utils.js [hunk 4]
```diff
@@ -249,6 +249,8 @@ module.exports = {
       var intervalJobs = [];
       var intervalTimes = [];
 
+      if(!itterations){ return callback(new Error('itterations is required')); }
```
Comment (gcoonrod) [inline initializers/utils.js]: Won't prevent someone passing in `Infinity` and creating an endless loop. But that would be a very specific action.

Comment (evantahler) [inline initializers/utils.js]: Yeah... That's their own fault then!

**Pull request merged**